src/badge.jsx

```diff
@@ -21,12 +20,39 @@ export default React.createClass({
     };
   },
   propTypes: {
+    /**
+     * This is the content rendered within the badge.
+     */
     badgeContent: React.PropTypes.node.isRequired,
+
+    /**
+     * Override the inline-styles of the badge element.
+     */
     badgeStyle: React.PropTypes.object,
+
+    /**
+     * The badge will be added relativelty to this node.
+     */
     children: React.PropTypes.node,
+
+    /**
+     * The css class name of the root element.
+     */
     className: React.PropTypes.string,
+
+    /**
+     * If true, the badge will use the primary badge colors.
+     */
     primary: React.PropTypes.bool,
+
+    /**
+     * If true, the badge will use the secondary badge colors.
+     */
     secondary: React.PropTypes.bool,
+
+    /**
+     * Override the inline-styles of the root element.
```

Comment:
I like your approach. be sure to remind me when I make this mistake again :+1: :+1: